autoload/vista/extension/rst.vim

```diff
@@ -0,0 +1,89 @@
+" Heavily inspired  from https://raw.githubusercontent.com/Shougo/unite-outline/master/autoload/unite/sources/outline/defaults/rst.vim
+
+let s:provider = fnamemodify(expand('<sfile>'), ':t:r')
+
+function! s:Execute() abort
+  let headers = []
+
+  let idx = 0
+  let lines = t:vista.source.lines()
+  let adornment_levels = {}
+  let adornment_id = 2
+
+  while idx < len(lines)
+    let line = lines[idx]
+    let matched_line = get(lines, idx + 1, '')
+    " Check the matching strictly.
+    if matched_line =~# '^\([[:punct:]]\)\1\{3,}$' && line !~# '^\s*$'
+        if idx > 1 && lines[idx - 1] == matched_line
+          " Title
+          let item = {'lnum': idx, 'text': l:line, 'level': 1}
+        else
+          " Sections
+          let item = {'lnum': idx, 'text': l:line}
+          let adchar = matched_line[0]
+          if !has_key(l:adornment_levels, adchar)
+          let l:adornment_levels[adchar] = l:adornment_id
+          let l:adornment_id += 1
+          endif
+          let item['level'] = l:adornment_levels[adchar]
+        endif
+        call add(headers, l:item)
+        let idx += 1
+    endif
+    let idx += 1
+ endwhile
+
+  return headers
+endfunction
+
+let s:default_icon = get(g:, 'vista_icon_indent', ['╰─▸ ', '├─▸ '])
```

Comment:
Move the consts on the top if possisble.

Reply:
done in 1c84ad7